### DISCUSSION TITLE
Why are deprecated linters included in Presets?

### DISCUSSION THREAD

**Discussion by bensaufley** [golangci/golangci-lint#2235] [Q&A] [answered]
I was encountering a deprecation warning for `scopelint` in my local config, which was based on the example config—despite not having invoked `scopelint` in my config. Took me a while to track down that the `bugs` preset includes `scopelint`, despite that warning.

I forked the repo to create a PR that would remove `scopelint` from the preset (especially considering `bugs` already includes the replacement package, `exportloopref`). To be clear I wasn't gonna remove `scopelint` from the repo, just from the preset—meaning people could still explicitly invoke it, but it wouldn't be implicitly included.

Then I realized that a lot of (all?) deprecated linters in the `lintersdb/manager.go` file are associated with presets. Why is that?

Seems like once they're deprecated, they shouldn't be in a preset, and should only be added when explicitly included. It doesn't seem like presets, which are controlled by the `golangci-lint` app, should automatically include packages that the app itself recognizes and warns as deprecated.

**Comment by ldez** [comment, marked as answer]
Hello,

There are 4 deprecated linters on 78 linters.

As we don't want to be breaking, those linters are not removed from the available linters and not removed from presets.

If you want to learn more about our deprecation cycle, you can read and participate to the following issue https://github.com/golangci/golangci-lint/issues/1987

**Reply by bensaufley** [reply]
Seems like something that could be solved with versioning but either way, is there a means of calling out why you're seeing the error and how to disable it? Or potentially add new presets that always exclude deprecated tools? It's unexpected behavior as a new user to run the recommended config and get deprecation warnings.

**Reply by ldez** [reply]
FYI presets are not recommended configuration, it's just linter categories.

**Reply by bensaufley** [reply]
Is that documented somewhere? There are two presets in the "example config" which, while I understand "example" ≠ "recommended," seems like the closest thing to a starter config that's available.

**Reply by Antonboom** [reply]
Continued [here](https://github.com/golangci/golangci-lint/issues/1987#issuecomment-1007490594)
